app/org/sagebionetworks/bridge/models/schedules/SurveyReference.java

```diff
@@ -1,58 +1,58 @@
 package org.sagebionetworks.bridge.models.schedules;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
 import java.util.Objects;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
 
 import org.joda.time.DateTime;
-import org.sagebionetworks.bridge.json.BridgeTypeName;
+import org.joda.time.format.ISODateTimeFormat;
+import org.sagebionetworks.bridge.config.BridgeConfigFactory;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 
 /**
  * This is a "soft" reference to a survey that does not need to include a createdOn timestamp. 
  * It can be used to create JSON for published versions of surveys as well as hard references 
  * to a specific version.
  */
-@BridgeTypeName("GuidCreatedOnVersionHolder")
 public final class SurveyReference {
 
-    public static final String SURVEY_RESPONSE_PATH_FRAGMENT = "/surveyresponses/";
-    public static final String SURVEY_PATH_FRAGMENT = "/surveys/";    private static final String PUBLISHED_FRAGMENT = "published";
-    private static final String REGEXP = "http[s]?\\://.*/surveys/([^/]*)/revisions/([^/]*)";
-    private static final Pattern patttern = Pattern.compile(REGEXP);
-    
-    public static final boolean isSurveyRef(String ref) {
-        return (ref != null && ref.matches(REGEXP));
-    }
+    private static final String BASE_URL = BridgeConfigFactory.getConfig().getBaseURL() + "/v3/surveys/";
     
+    private final String identifier;
     private final String guid;
     private final DateTime createdOn;
     
-    public SurveyReference(String ref) {
-        checkNotNull(ref);
-        Matcher matcher = patttern.matcher(ref);
-        matcher.find();
-        this.guid = matcher.group(1);
-        String createdOnString = (PUBLISHED_FRAGMENT.equals(matcher.group(2))) ? null : matcher.group(2);
-        this.createdOn = (createdOnString != null) ? DateTime.parse(createdOnString) : null;    
-        checkNotNull(guid);
+    @JsonCreator
+    public SurveyReference(@JsonProperty("identifier") String identifier, @JsonProperty("guid") String guid,
+                    @JsonProperty("createdOn") DateTime createdOn) {
+        this.identifier = identifier;
```

Comment:
The activity validator is checking that the reference objects are all properly constructed, again so we can return one exception with all the errors in the object. I should limit the constructor though so only the activity builder can construct these (package private).